ui/dashboards/src/views/ViewDashboard.tsx

```diff
@@ -27,9 +29,45 @@ export interface ViewDashboardProps extends BoxProps {
 export function ViewDashboard(props: ViewDashboardProps) {
   const { dashboardResource, sx, children, ...others } = props;
 
+  const { isEditMode, setEditMode } = useEditMode();
+
+  const onEditButtonClick = () => {
+    setEditMode(true);
+  };
+
+  const onCancelButtonClick = () => {
+    setEditMode(false);
+  };
+
   return (
     <TimeRangeStateProvider initialValue={{ pastDuration: dashboardResource.spec.duration }}>
       <TemplateVariablesProvider variableDefinitions={dashboardResource.spec.variables}>
+        {isEditMode && (
+          <Toolbar sx={{ display: 'flex', paddingLeft: '16px', paddingRight: '16px' }}>
+            <Typography variant="h6" sx={{ marginRight: '1rem' }}>
+              Edit Dashboard
+            </Typography>
+            <Box sx={{ marginLeft: 'auto' }}>
+              <Button
+                variant="outlined"
+                sx={{
+                  margin: '0 4px',
+                }}
+                onClick={onCancelButtonClick}
+              >
+                Cancel
+              </Button>
+              <Button
+                variant="contained"
+                sx={{
+                  margin: '0 4px',
+                }}
+              >
+                Save
+              </Button>
+            </Box>
```

Comment:
Nit/suggestion: might consider MUI `Stack` here for the spacing between buttons as an alternative to the custom styles.